src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1975,12 +1990,18 @@ protected override void Validate(object arguments, EngineIntrinsics engineIntrin
                         Metadata.ValidateNotNullOrEmptyFailure);
                 }
             }
-            else if ((ienum = arguments as IEnumerable) != null)
+            else if (IsArgumentCollection(arguments.GetType(), out bool isElementValueType))
```

Comment:
No, the check of element type doesn't involve inspecting the elements of the collection. You can see that the type of `arguments` is passed in instead of the `arguments` itself.
Please see the constructor of [`ParameterCollectionTypeInformation`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/CompiledCommandParameter.cs#L613) for details.